### ISSUE TITLE
Switching to float for histogram building

### ISSUE THREAD

**Issue by Shiki-H** [dmlc/xgboost#10913]
Hi, I'd like to get some clarifications about the decision to stick to double for `GHistRow` that is used in histogram building. @RAMitchell mentioned in [this](https://github.com/hcho3/xgboost-fast-hist-perf-lab/issues/15#issuecomment-441519612) comment that switching to float will result in significant accuracy degradation. I'm just wondering how is performance measured in this case. More generally, if say we have ways to improve the runtime of model building that would lead to changes in model metrics, from the development team's perspective, what level of variation is acceptable?

**Comment by trivialfis** [comment]
Hi, XGBoost GPU used to have this option. I proposed and merged the removal of this option (a long time ago). The model wasn't converging when f32 was used for accumulation, and the number of samples was large. For normal regression, this can be obvious since one would see metrics like `rmse` increasing for the training dataset. But it's not obvious if the metric is in a log scale like logistic regression. In that case, even if the metric is somewhat improving, the model can be complete garbage.

>  what level of variation is acceptable

As a result, it's not about variation, it's about converging or not, and whether a user can easily tell he or she is running into trouble.

**Comment by trivialfis** [comment]
Feel free to close if there are no further questions.

**Comment by Shiki-H** [comment]
@trivialfis thanks, that was very helpful
